### ISSUE TITLE
🐛 `useButtonType` warning when `type` is provided

### ISSUE THREAD

**Issue by o-az** [biomejs/biome#4072]
### Environment information

[playground link](https://biomejs.dev/playground/?code=UgBlAGEAYwB0AC4AYwByAGUAYQB0AGUARQBsAGUAbQBlAG4AdAAoACIAYgB1AHQAdABvAG4AIgAsACAAewAgAHQAeQBwAGUAOgAgACIAYgB1AHQAdABvAG4AIgAgAH0ALAAgACIAZgBvAG8AIgApAAoACgBSAGUAYQBjAHQALgBjAHIAZQBhAHQAZQBFAGwAZQBtAGUAbgB0ACgAIgBiAHUAdAB0AG8AbgAiACwAIAB7ACAAdAB5AHAAZQA6ACAAIgBzAHUAYgBtAGkAdAAiACAAfQAsACAAIgBmAG8AbwAiACkACgA%3D)

```block
CLI:
  Version:                      1.9.2
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           linux

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  NO_COLOR:                     unset
  TERM:                         "xterm-ghostty"
  JS_RUNTIME_VERSION:           unset
  JS_RUNTIME_NAME:              unset
  NODE_PACKAGE_MANAGER:         unset

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    true
  VCS disabled:                 false

Workspace:
  Open Documents:               0
```

### What happened?

```ts
React.createElement("button", { type: "button" }, "foo")
// Provide a valid type prop for the button element.biomelint/a11y/useButtonType
```

### Expected result

no `element.biomelint/a11y/useButtonType` warning when `type` is provided

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct
